src/mlpack/methods/reinforcement_learning/environment/pendulum.hpp

```diff
@@ -140,6 +154,15 @@ class Pendulum
         -maxAngularVelocity), maxAngularVelocity);
     nextState.Theta() = theta + newAngularVelocity * dt;
 
+    // Check if the episode has terminated
+    bool done = IsTerminal(nextState);
+
+    // Do not reward the agent if time ran out.
+    if (done && maxTimeSteps != 0 && timeStepsPerformed >= maxTimeSteps)
```

Comment:
Should we terminate the moment we reached maximum iterations allowed? Let me know your thoughts on this?

Reply:
Oh, I see you are doing the same. I thought done is being taken from angle terminal condition. Ignore the above comment.